### DISCUSSION TITLE
Multi-objective Particle Swarm Optimization in Pymoo?

### DISCUSSION THREAD

**Discussion by Freestyleyang** [anyoptimization/pymoo#555] [Q&A]
Hi There,

Thank you for developing such a wonderful package for optimization!

I wonder if it is possible to include Multi-objective Particle Swarm Optimization (**MOPSO**) as an algorithm in Pymoo? MOPSO is powerful in optimization and it would be great if Pymoo could include it. Thanks.

**Comment by blankjul** [comment]
My research has never been directly focused on PSO and this is why I have not spent much time on the implementations.

Would you be willing to implement MOPSO and contribute to pymoo?

**Reply by Freestyleyang** [reply]
@blankjul Thank you for your reply. I just notice that MOPSO is implemented in the Python package "[pyMultiobjective](https://github.com/Valdecy/pyMultiobjective)". I wonder if it is possible to incorporate MOPSO into pymoo through pyMultiobjective? Thanks.

**Comment by blankjul** [comment]
I just looked at the framework and calling it directly would probably be an issue because of the GNU license. However, the algorithm could be reimplemented based on their implementation and the paper. 

Are you planning to do research in this area? I have tried to find someone implemented any type of MO PSO algorithm in the past but was not successful. Are you willing to work on a PR?

```
[Apache's position](https://www.apache.org/licenses/GPL-compatibility.html) is that their v2.0 license is compatible with GPL v3.0 but only in the forward direction. That is, you can embed Apachev2-licensed software in GPL v3.0 licensed projects, but not the other way around.
```
https://opensource.stackexchange.com/questions/11082/why-is-the-gnu-gpl-v3-compatible-with-the-apache-license-v2-0
